fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadPendingTask.java

```diff
@@ -0,0 +1,412 @@
+                partitionIds = tableIdToPartitionIds.get(tableId);
+            } else {
+                partitionIds = Sets.newHashSet();
+                tableIdToPartitionIds.put(tableId, partitionIds);
+            }
+
+            Set<Long> groupPartitionIds = aggKey.getPartitionIds();
+            // if not assign partition, use all partitions
+            if (groupPartitionIds == null || groupPartitionIds.isEmpty()) {
+                for (Partition partition : table.getPartitions()) {
+                    partitionIds.add(partition.getId());
+                }
+
+                allPartitionsTableIds.add(tableId);
+            } else {
+                partitionIds.addAll(groupPartitionIds);
+            }
+        }
+    }
+
+    private Map<String, EtlColumn> createEtlColumnMap(OlapTable table) {
+        Map<String, EtlColumn> nameToColumnMap = Maps.newHashMap();
+        for (Column column : table.getBaseSchema()) {
+            nameToColumnMap.put(column.getName(), createEtlColumn(column));
+        }
+        return nameToColumnMap;
+    }
+
+    private Map<Long, EtlIndex> createEtlIndexes(OlapTable table) throws LoadException {
+        Map<Long, EtlIndex> idToEtlIndex = Maps.newHashMap();
+
+        for (Map.Entry<Long, List<Column>> entry : table.getIndexIdToSchema().entrySet()) {
+            long indexId = entry.getKey();
+            int schemaHash = table.getSchemaHashByIndexId(indexId);
+
+            List<EtlColumn> columnMaps = Lists.newArrayList();
+            for (Column column : entry.getValue()) {
+                columnMaps.add(createEtlColumn(column));
+            }
+
+            List<String> distributionColumnRefs = Lists.newArrayList();
+            DistributionInfo distributionInfo = table.getDefaultDistributionInfo();
+            switch (distributionInfo.getType()) {
+                case HASH:
+                    HashDistributionInfo hashDistributionInfo = (HashDistributionInfo) distributionInfo;
+                    for (Column column : hashDistributionInfo.getDistributionColumns()) {
+                        distributionColumnRefs.add(column.getName());
+                    }
+                    break;
+                case RANDOM:
+                    for (Column column : entry.getValue()) {
+                        if (!column.isKey()) {
+                            break;
+                        }
+                        distributionColumnRefs.add(column.getName());
+                    }
+                    break;
+                default:
+                    LOG.warn("unknown distribution type. type: {}", distributionInfo.getType().name());
+                    throw new LoadException("unknown distribution type. type: " + distributionInfo.getType().name());
+            }
+
+            idToEtlIndex.put(indexId, new EtlIndex(columnMaps, distributionColumnRefs, schemaHash));
+        }
+
+        return idToEtlIndex;
+    }
+
+    private EtlColumn createEtlColumn(Column column) {
```

Comment:
I think it's better to add toString method here for LOG ,because it's hard to remote debug SparkEtlJob.